### PR TITLE
Update NextJs gitignore

### PR DESCRIPTION
**Pull request by mriios**
## New or update

Select the appropriate check box for this pull request. This helps when merging to ensure there are no conflicts with other templates or misunderstandings of how thee template list works.

### New

- [ ] Template - New `.gitignore` template
- [ ] Composition - Template made from smaller templates
- [ ] Inheritance - Template similar to an existing template
- [ ] Patch - Template extending functionality of existing template

### Update

- [x] Template - Update existing `.gitignore` template

## Details
Update to NextJS gitignore, including default files and routes.

### CODE REVIEW

File: templates/NextJS.gitignore [hunk 1]
```diff
@@ -1,2 +1,33 @@
...
+
+# Testing
+/coverage
+
+# Next.js
+/.next/
+/out/
+
+# Production
+/Build
```
Comment (filipechagas) [inline templates/NextJS.gitignore:14]:
```suggestion
/build
```

File: templates/NextJS.gitignore [hunk 2]
```diff
@@ -1,2 +1,33 @@
...
+.pnpm-debug.log*
+
+# Local env files
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+# Vercel
+.vercel
```
Comment (filipechagas) [inline templates/NextJS.gitignore:33]:
```suggestion
.vercel

#editors
.idea
.vscode
```

Comment (filipechagas) [reply templates/NextJS.gitignore:33]: Can you also please fix missing EOL?

**Review by filipechagas — CHANGES_REQUESTED** [review]
Hi and thank you for contributing.

I've left a few suggestions.

File: templates/NextJS.gitignore (hunk 2 above)
Comment (mriios) [reply templates/NextJS.gitignore:33]: @filipechagas do we want to add the editors even if they're in their own gitignore?

Comment (filipechagas) [reply templates/NextJS.gitignore:33]: It was just a suggestion, as you may usually find these entries in NextJS gitignore files. But I agree with you. Lets leave the editors out.

Please fix EOL and we're good to merge 🙂

**Review by filipechagas — APPROVED** [review]

**Pull request merged**